src/batch/src/executor/insert.rs

```diff
@@ -96,9 +96,9 @@ impl InsertExecutor {
 
             // No need to check for duplicate columns. This is already validated in binder
             if !&self.column_idxs.is_sorted() {
-                let mut ordered_cols: Vec<Column> = Vec::with_capacity(len);
-                for idx in &self.column_idxs {
-                    ordered_cols.push(columns[*idx].clone());
+                let mut ordered_cols: Vec<Column> = columns.clone();
+                for (i, idx) in self.column_idxs.iter().enumerate() {
+                    ordered_cols[*idx] = columns[i].clone()
```

Comment:
There is a method `pub fn reorder_columns(self, column_mapping: &[usize]) -> Self` of `DataChunk`. I think it is exactly what you need here.

Reply:
I tried to use it before, but ran into an error: `thread '<unnamed>' panicked at 'insertion index (is 1) should be <= len (is 0)', library/alloc/src/vec/mod.rs:1444:13`. I am under the impression that `reoder_columns` deletes something here that I still wanna keep